src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2237,6 +2263,8 @@ internal static bool CheckValEscape(SyntaxNode node, BoundExpression expr, uint
                 case BoundKind.DefaultExpression:
                 case BoundKind.Parameter:
                 case BoundKind.ThisReference:
+                case BoundKind.TupleLiteral:
+                case BoundKind.ConvertedTupleLiteral:
                     // always returnable
                     return true;
```

Comment:
>true [](start = 27, length = 4)

I don't think this is correct for a tuple literal. For example, I could write
``` c#
(tupleA, someString) = (tupleB, null)
```
This is all valid (because the right-hand-side has no type), but the safe-to-escape value computed from the right-hand-side is from the safe-to-escape of the variable `tupleB`, and the whole thing is only valid if the assignment `tupleA = tupleB` would be valid.

Reply:
I corrected the comment just above.

Reply:
Thanks for the scenario.
I don't agree with your expectation though. The tuple on the right does receive a type, by inferring a merged type from both sides. In this case, the RHS will be made into a `(Span, string)` which is illegal.
So `(spanA, someString) = (spanB, null);` should produce an constraint error (Span cannot be used as a type argument).